### PR TITLE
✨Add Artsy API

### PR DESCRIPTION
**Pull request by alihaiderrizvi**
The public API doesn't have access to all items on Artsy so maybe we might have to move to a partner API in the near future.

This version of the code gets the following data:

- Artist details
- Artwork details
- Show details
- Profile details

### CODE REVIEW

File: scripts/enrichment.py [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
+from fetch_imdb import *
+from fetch_goodreads import *
+from fetch_artsy import *
+
+APIs = ["www.goodreads.com", "www.imdb.com", "open.spotify.com", "www.artsy.net"]
+
+
+def enrich_test(url):
+    site_name = url.split("/")[2]
+    print(site_name)
```
Comment (hxghhhh) [inline scripts/enrichment.py:13]: remove this print, unless you want it for logging purposes. If you want it write a better message so know exactly where to trace the log back to

File: scripts/enrichment.py [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
+from bs4 import BeautifulSoup, SoupStrainer
+from fetch_spotify import spotify_get
+from fetch_imdb import *
+from fetch_goodreads import *
+from fetch_artsy import *
+
+APIs = ["www.goodreads.com", "www.imdb.com", "open.spotify.com", "www.artsy.net"]
+
+
+def enrich_test(url):
```
Comment (hxghhhh) [inline scripts/enrichment.py:11]: Can you write a test for this to make sure all edge cases are handle. I can send you a boilerplate for `pytest` if you've never written a test

Comment (hxghhhh) [inline scripts/enrichment.py:11]: Also can we add typing here?

https://docs.python.org/3/library/typing.html

File: scripts/fetch_artsy.py [hunk 3]
```diff
@@ -1,49 +1,58 @@
...
-    elif artsy_type == 'search': #
-        pass
-    elif artsy_type == 'shows': #
-        pass
-    else:
-        return api_data
+import requests
+
+
+def fetch_artsy(url):
```
Comment (hxghhhh) [inline scripts/fetch_artsy.py:4]: add typing

File: scripts/fetch_artsy.py [hunk 4]
```diff
@@ -1,49 +1,58 @@
...
+
+    client_id = '77e9422bdf38d3bc0179'
```
Comment (hxghhhh) [inline scripts/fetch_artsy.py:6]: nit: move these to a constants.py file to make easier to refactor later on also centralizes the string so we can import in other places

File: scripts/fetch_artsy.py [hunk 5]
```diff
@@ -1,49 +1,58 @@
...
+            api_data.update(requests.get(api_data['_links']['artworks']['href'], headers=headers).json())
+        except:
+            pass
+    
+    elif artsy_type == 'artwork':
+        target = starting_url + '/artworks/' + artsy_id
+        api_data = requests.get(target, headers = headers).json()
+        try:
+            api_data.update(requests.get(api_data['_links']['artists']['href'], headers=headers).json())
+        except:
```
Comment (hxghhhh) [inline scripts/fetch_artsy.py:39]: we should have a log here saying when the api files, using `pass` is definitely a bad pattern

@joelrojo do we have any observability setup up to help the young bulls monitor things when they 🚢

File: scripts/fetch_artsy.py [hunk 6]
```diff
@@ -1,49 +1,58 @@
...
+            pass
+    
+    elif artsy_type == 'show':
+        artsy_id = artsy_id.split('?')[0]
+        target = starting_url+'/shows/'+artsy_id
+        api_data = requests.get(target, headers = headers).json()
+        try:
+            api_data.update(requests.get(api_data['_links']['images']['href'], headers=headers).json())
+        except:
+            pass
```
Comment (hxghhhh) [inline scripts/fetch_artsy.py:49]: same as other pass comment

File: scripts/fetch_artsy.py [hunk 7]
```diff
@@ -1,49 +1,58 @@
...
+        'X-Xapp-Token': auth['token']
+    }
+
+    if artsy_type == 'artist':
+        target = starting_url+'/artists/'+artsy_id
+        api_data = requests.get(target, headers=headers).json()
+        try:
+            api_data.update(requests.get(api_data['_links']['artworks']['href'], headers=headers).json())
+        except:
+            pass
```
Comment (hxghhhh) [inline scripts/fetch_artsy.py:32]: same as other pass comment

**Review by hxghhhh — CHANGES_REQUESTED** [review]

File: scripts/fetch_artsy.py (hunk 4 above)
Comment (joelrojo) [reply scripts/fetch_artsy.py:6]: @hxghhhh agreed. Also think we can use [dotenv](https://github.com/theskumar/python-dotenv) to keep these off of the repo

File: scripts/fetch_artsy.py (hunk 5 above)
Comment (joelrojo) [reply scripts/fetch_artsy.py:39]: @hxghhhh not yet, but @FaazAbidi is setting up sentry so should have some basic optics into errors at least. You have any recommendations on what to integrate?

**Comment by joelrojo** [comment]
Closing this since it's superseded by PR #3 

@alihaiderrizvi @FaazAbidi in the future you can just keep pushing onto this remote branch and it'll automatically update the PR. That way you don't have migrate everything to a new PR and we can keep things all in one place (history of comments, updates etc)

It's also a good habit to resolve comments once you're done and then tag us in a new comment once the PR is ready to review again.

Great work so far!
